bundles/admin/admin-announcements/view/AnnouncementsForm.jsx

```diff
@@ -0,0 +1,163 @@
+import React from "react";
+import PropTypes from 'prop-types';
+import { Form, Input, Button, Switch, Row } from "antd";
+import { Message, Confirm, DateRange } from 'oskari-ui';
+import { Controller, LocaleConsumer } from 'oskari-ui/util';
+import moment from 'moment';
+
+/*
+This file contains the form for admin-announcements.
+This is the main file for creating and editing announcements.
+*/
+const { TextArea } = Input;
+const rangeConfig = {
+  rules: [
+    {
+      type: "array",
+      required: true,
+      message: <Message messageKey='dateError' />
+    }
+  ]
+};
+
+const DATEFORMAT = 'YYYY-MM-DD';
+
+const AnnouncementsForm = ({controller,  title, key, form, index}) => {
+
+  const onFinish  = fieldsValue => {
+    // Should format date value before submit.
+    const rangeValue = fieldsValue["range_picker"];
+
+    const values = {
+      title: fieldsValue["title"],
+      content: fieldsValue["content"],
+      begin_date: rangeValue[0].format(DATEFORMAT), 
+      end_date: rangeValue[1].format(DATEFORMAT),
+      active: fieldsValue["active"]
+    };
+
+    if (form.id === undefined) {
+      controller.saveAnnouncement(values);
+    } else {
+      values.id = form.id;
+      controller.updateAnnouncement(values);
+    }
+  }
+  
+  //Set initial values to date range depending on if we are editing or creating an announcement
+  const rangeInitial = () => {
+
+    if (form.begin_date && form.end_date) {
+      return [moment(form.begin_date, DATEFORMAT), moment(form.end_date, DATEFORMAT)]; 
+    } else {
+      return [moment(moment(),DATEFORMAT), moment(moment(),DATEFORMAT)];  
+    }
+
+  } 
+
+  //Active value set depending on if creating a new announcement or editing an old one
+  const activeInitial = () => {
+    if(form.active === undefined) {
+      return true;
+    } else {
+      return form.active;
+    }
+  }
+
+    return (
+      <div>
+            <Form layout="vertical" 
+              onFinish={onFinish} 
+              initialValues={{
+                title: form.title !== undefined ? form.title : title,
+                content: form.content,
+                range_picker: rangeInitial(),
+                active: activeInitial(),
+              }}>
+              <Form.Item
+                name="title"
+                label={<Message messageKey='newAnnouncement.title' />}
+                rules={[
+                  {
+                    required: true,
+                    message: <Message messageKey='titleError' />,
+                    whitespace: true
+                  }
+                ]}
+                value={title}
+                onChange={ evt => controller.updateTitle(evt.target.value)}
```

Comment:
I'm not sure you need to do this onChange() with the antd form component. Maybe try if removing it will function the same? And if so just remove it from the controller.